### DISCUSSION TITLE
Set Correct Time Zone for Recording Folders

### DISCUSSION THREAD

**Discussion by ghost** [blakeblackshear/frigate#7695] [Ask A Question] [answered]
hi !
I just realized that the recording subfolders (under "recordings" folder) are not correct with my timezone (here in France)

The format of the daily folder is correct, the subfolders (by time obviously) are not.
  a screenshots will be more telling: https://i.goopics.net/ysa98s.jpg

I saw that in the "docker-compose.yml" that the /etc/localtime is get from the host machine:
```
volumes:
       - /etc/localtime:/etc/localtime:ro
```
which should therefore return mine which is `/etc/localtime -> /usr/share/zoneinfo/Europe/Paris`

Is there an option/config in "config.yml" that could change this?

Thanks in advance

**Comment by NickM-27** [comment]
Folder structure is done with UTC time to avoid issues with daylight savings time among other advantages. This is not configurable, the recordings files are considered internal metadata and should not be interacted with directly -- that is what the webui and API are for.

**Reply by ghost** [reply]
humm as they would say back home: “that’s a Norman answer”

I don't master everything, far from that, but honestly I don't see how using the local timezone influences anything.
we don't change a timezone every morning, for fun, it's normally transparent for any application, it formats data, folder names, file names or anything else, it does it on the date/locale not on a hard-fixed time zone

you tell me, "that's what the webui and the API are for"... uhhh yes, maybe, I don't know, but it's also simple/quick to read the files in a tree structure files, right?

because... if I look for recorded videos of a date and time, the date is ok, but the time, I have to say to myself "ah yes but H-2.."

once again I don't understand (the sentence yes) what this notion of "metadata" does for a tree structure.

To understand this "problem" (because it seems to me to be one) I looked and in fact you "force" UTC (line 326 of maintainer.py, little friendly wink, line 334 it's useless to do)
I haven't watched the bd, but I feel it's the same ;)

friendly, I need more than the notion of "internal metadata" ;) because once again it has nothing internal... it's an external tree
When it comes to Daylight, I don't even understand 8 a.m. is 8 p.m. (!)
I really want to be convinced of the reason behind this, but that's not the case at the moment ;)

Sincerely friendly response

**Reply by NickM-27** [reply]
> once again I don't understand (the sentence yes) what this notion of "metadata" does for a tree structure.

It means that it is all data for frigate to do its job, it is internal to frigate and not meant to be interacted with directly by the user. It is under the hood data that frigate uses to present data to the user via the WebUI and API.

> When it comes to Daylight, I don't even understand 8 a.m. is 8 p.m. (!)

I would suggest reading about daylight savings time: https://en.wikipedia.org/wiki/Daylight_saving_time

This type of change that happens twice a year can cause recording footage to be overwritten. 

Using UTC for the file structure also gives a common starting point for frigate to convert to different timezones, which powers things like the UI timezone config that was added in 0.12 alongside this change.

**Comment by ghost** [comment]
> It means that it is all data for frigate to do its job, it is internal to frigate and not meant to be interacted with directly by the user. It is under the hood data that frigate uses to present data to the user via the WebUI and API.

> I would suggest reading about daylight savings time: https://en.wikipedia.org/wiki/Daylight_saving_time
> This type of change that happens twice a year can cause recording footage to be overwritten.

One point for u :)  in France this notion was to disappear in 2021, "because of" Covid 19 obliges (again) it is postponed at least not before 2024 .

 
> Using UTC for the file structure also gives a common starting point for frigate to convert to different timezones, which powers things like the UI timezone config that was added in 0.12 alongside this change.

Clear.. now I am convinced and understand the reasons, it is my use (or habit) of preferring to manage directly via files rather than via a UI which is my "problem"

So this raises a question for me:
via the UI, it is possible to download the "clips" of the events (grouping of 10s event videos from folders) OK
Is it possible directly in frigate to have the "clips" (without having to download them manually)
it is possible to do it externally via ffmpeg cron or other but if it is provided in frigate, how to activate it, I do not seem to have seen this notion in the Doc, because suddenly, the 10s segments would become useless. ..

Thanks

**Reply by NickM-27** [reply, marked as answer]
> Is it possible directly in frigate to have the "clips" (without having to download them manually)

I don't know what you mean by this, the UI exists to watch clips as well as watch 24/7 recordings

> it is possible to do it externally via ffmpeg cron or other but if it is provided in frigate, how to activate it, I do not seem to have seen this notion in the Doc, because suddenly, the 10s segments would become useless. ..

yes of course, that is what the API is for. Here are some links to useful APIs

[Get recording for specific time frame](https://docs.frigate.video/integrations/api#get-vodcamerastartstart-timestampendend-timestampindexm3u8)

[Get recording for a specific hour](https://docs.frigate.video/integrations/api#get-vodyear-monthdayhourcameramasterm3u8)

[Get recording for specific event](https://docs.frigate.video/integrations/api#get-vodeventevent-idindexm3u8)

**Reply by ghost** [reply]
I admit that I have not, for the moment, looked at the API side...
there are so many things to integrate into Frigate, that I am focused on the "basic" configs

Thank you very much for the time given, it is very much appreciated
